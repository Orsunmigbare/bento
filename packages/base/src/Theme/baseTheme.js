import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
