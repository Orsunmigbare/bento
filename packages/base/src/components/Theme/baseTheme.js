export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
