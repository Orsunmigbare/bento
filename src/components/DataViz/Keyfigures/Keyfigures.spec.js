import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import KeyFigures from "./Keyfigures"

describe("KeyFigures", () => {
  describe("Snapshots", () => {
    it("Empty Array", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <KeyFigures figures={[]} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("With figures", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <KeyFigures
              figures={[
                {
                  label: "Label",
                  amount: 45854,
                  key: 546456 + "booked"
                },
                {
                  label: "Available",
                  amount: 456486,
                  key: 56456 + "available"
                },
                {
                  label: "New Label",
                  amount: 346,
                  key: 364 + "available"
                }
              ]}
            />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("With one figure", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <KeyFigures
              figures={[
                {
                  label: "Label",
                  amount: 45854,
                  key: 546456 + "booked"
                }
              ]}
            />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("With children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <KeyFigures
              figures={[
                {
                  label: "Label",
                  amount: 45854,
                  key: 546456 + "booked"
                },
                {
                  label: "Available",
                  amount: 456486,
                  key: 56456 + "available"
                },
                {
                  label: "New Label",
                  amount: 346,
                  key: 364 + "available"
                }
              ]}
            >
              <p>Hey</p>
            </KeyFigures>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
