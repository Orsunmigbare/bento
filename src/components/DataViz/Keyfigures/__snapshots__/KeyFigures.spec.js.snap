// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyFigures Snapshots Empty Array 1`] = `
<ul
  className="sc-AykKC hEeApa"
/>
`;

exports[`KeyFigures Snapshots With children 1`] = `
<ul
  className="sc-AykKC hEeApa"
>
  <li
    className="sc-AykKD jubovb"
  >
    <h3
      className="sc-AykKE cOBxxr"
    >
      Label
    </h3>
    <span
      className="sc-AykKF iTgtqP"
    >
       
      45 854
    </span>
  </li>
  <li
    className="sc-AykKD jubovb"
  >
    <h3
      className="sc-AykKE cOBxxr"
    >
      Available
    </h3>
    <span
      className="sc-AykKF iTgtqP"
    >
       
      456 486
    </span>
  </li>
  <li
    className="sc-AykKD jubovb"
  >
    <h3
      className="sc-AykKE cOBxxr"
    >
      New Label
    </h3>
    <span
      className="sc-AykKF iTgtqP"
    >
       
      346
    </span>
  </li>
</ul>
`;

exports[`KeyFigures Snapshots With figures 1`] = `
<ul
  className="sc-AykKC hEeApa"
>
  <li
    className="sc-AykKD jubovb"
  >
    <h3
      className="sc-AykKE cOBxxr"
    >
      Label
    </h3>
    <span
      className="sc-AykKF iTgtqP"
    >
       
      45 854
    </span>
  </li>
  <li
    className="sc-AykKD jubovb"
  >
    <h3
      className="sc-AykKE cOBxxr"
    >
      Available
    </h3>
    <span
      className="sc-AykKF iTgtqP"
    >
       
      456 486
    </span>
  </li>
  <li
    className="sc-AykKD jubovb"
  >
    <h3
      className="sc-AykKE cOBxxr"
    >
      New Label
    </h3>
    <span
      className="sc-AykKF iTgtqP"
    >
       
      346
    </span>
  </li>
</ul>
`;

exports[`KeyFigures Snapshots With one figure 1`] = `
<ul
  className="sc-AykKC hEeApa"
>
  <li
    className="sc-AykKD jubovb"
  >
    <h3
      className="sc-AykKE cOBxxr"
    >
      Label
    </h3>
    <span
      className="sc-AykKF iTgtqP"
    >
       
      45 854
    </span>
  </li>
</ul>
`;
